edgelet/iotedge/src/check/mod.rs

```diff
@@ -375,25 +217,62 @@ impl Check {
         }))
     }
 
+    fn checks() -> [(&'static str, Vec<Box<dyn Checker>>); 2] {
+        /* Note: keep ordering consistant. Later tests may depend on earlier tests. */
+        [
+            (
+                "Configuration checks",
+                vec![
+                    Box::new(WellFormedConfig::default()),
+                    Box::new(WellFormedConnectionString::default()),
+                    Box::new(ContainerEngineInstalled::default()),
+                    Box::new(WindowsHostVersion::default()),
+                    Box::new(Hostname::default()),
+                    Box::new(IotedgedVersion::default()),
+                    Box::new(HostLocalTime::default()),
+                    Box::new(ContainerLocalTime::default()),
+                    Box::new(ContainerEngineDns::default()),
+                    Box::new(ContainerEngineIPv6::default()),
+                    Box::new(IdentityCertificateExpiry::default()),
+                    Box::new(CertificatesQuickstart::default()),
+                    Box::new(ContainerEngineIsMoby::default()),
+                    Box::new(ContainerEngineLogrotate::default()),
+                    Box::new(EdgeAgentStorageMounted::default()),
+                    Box::new(EdgeHubStorageMounted::default()),
+                ],
+            ),
+            ("Connectivity checks", {
+                let mut tests: Vec<Box<dyn Checker>> = Vec::new();
+                tests.push(Box::new(HostConnectDpsEndpoint::default()));
+                tests.extend(get_host_connect_iothub_tests());
+                tests.extend(get_host_container_iothub_tests());
+                tests
+            }),
+        ]
+    }
+
     pub fn possible_ids() -> impl Iterator<Item = &'static str> {
-        CHECKS
+        let result: Vec<&'static str> = Check::checks()
             .iter()
-            .flat_map(|(_, section_checks)| *section_checks)
-            .map(|(check_id, _, _)| *check_id)
+            .flat_map(|(_, section_checks)| section_checks)
+            .map(|check| check.id())
+            .collect();
+
+        result.into_iter()
     }
 
     pub fn print_list() -> Result<(), Error> {
         // All our text is ASCII, so we can measure text width in bytes rather than using unicode-segmentation to count graphemes.
-        let widest_section_name_len = CHECKS
+        let widest_section_name_len = Check::checks()
```

Comment:
Just get the checks once with `let checks = Check::checks();` , then use this same `checks` for `widest_section_name_len`, `widest_check_id_len` and the actual for-loop.